lang/c++/impl/json/JsonIO.hh

```diff
@@ -116,6 +116,8 @@ public:
         return curToken;
     }
 
+    bool hasMore();
```

Comment:
Is it possible not to add `hasMore()` by reusing `peek()` above?

Reply:
It is possible. But the reason I did not do that is because `peek()` throws an exception if it's at the end, so it seemed better to add a way to check for more instead of trying to peek and catching. Also the current exception is just an `Exception`, not something like `EOFException`, so the code would need to actually inspect the string message to make sure it is EOF, which seemed brittle and undesirable.